src/mame/misc/toysoldier.cpp

```diff
@@ -0,0 +1,148 @@
+spanish:
+https://youtu.be/na70qV8awqE
+english female:
+https://youtu.be/wHEyHoKZkJk
+english male:
+https://youtu.be/aArV8y5TuzM?t=13
+amusements 
+https://youtu.be/qUiQgHpiAyI
+*/
+#include "emu.h"
+#include "cpu/mcs51/mcs51.h"
+#include "sound/ay8910.h"
+#include "sound/ymopl.h"
+#include "speaker.h"
+
+namespace {
+
+class toysoldier_state : public driver_device
+{
+public:
+	toysoldier_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		
+	{ }
+
+	void toysoldier(machine_config &config);
+
+private:
+	void ay_port_a_w(uint8_t data);
+	void ay_port_b_w(uint8_t data);
+	void io_map(address_map &map);
+	void program_map(address_map &map);
+
+	virtual void machine_start() override;
+
+
+};
+
+static INPUT_PORTS_START( toysoldier )
+    PORT_START("IN1") //  Gantry. 
+    PORT_BIT(0x08, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_1)
+	PORT_BIT(0x10, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_2)
+	PORT_BIT(0x20, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_3)
+    PORT_START("IN2")
+    PORT_BIT(0x80, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_Q)
+	PORT_BIT(0x40, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_W)
+	PORT_BIT(0x20, IP_ACTIVE_LOW, IPT_BUTTON1) 
+	PORT_BIT(0x10, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_R)
+	PORT_BIT(0x08, IP_ACTIVE_LOW, IPT_JOYSTICK_RIGHT) 
+	PORT_BIT(0x04, IP_ACTIVE_LOW, IPT_JOYSTICK_DOWN) 
+	PORT_BIT(0x02, IP_ACTIVE_LOW, IPT_JOYSTICK_UP) 
+	PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_JOYSTICK_LEFT)
+	PORT_START("IN3") // coin
+	PORT_BIT(0x80, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_A)
+	PORT_BIT(0x40, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_S)
+	PORT_BIT(0x20, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_D)
+	PORT_BIT(0x10, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_F)
+	PORT_BIT(0x08, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_G)
+	PORT_BIT(0x04, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_H)
+	PORT_BIT(0x02, IP_ACTIVE_LOW, IPT_COIN1) 
+	PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_COIN2)
+    PORT_START("DSW1")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW0:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW0:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW0:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW0:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW0:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW0:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW0:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW0:8")
+
+	PORT_START("DSW2")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW1:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW1:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW1:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW1:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW1:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW1:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW1:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW1:8")
+	
+
+INPUT_PORTS_END
+
+
+void toysoldier_state::program_map(address_map &map)
+{
+	map(0x0000, 0xffff).rom();
+}
+void toysoldier_state::io_map(address_map &map)
```

Comment:
It would be good to skip a line here.

I think you got the idea. Try to better organize the line-spacing in your code to be more consistent.